internal/client/client.go

```diff
@@ -868,6 +868,87 @@ func (a *AuthApiClient) ListResourcePermissions() ybmclient.ApiListResourcePermi
 	return a.ApiClient.AuthApi.ListResourcePermissions(a.ctx)
 }
 
+func (a *AuthApiClient) GetSensitivePermissions() (map[string][]string, error) {
+	resourcePermissionsResp, resp, err := a.ListResourcePermissions().Execute()
+	if err != nil {
+		c, _ := httputil.DumpResponse(resp, true)
+		logrus.Debug(c)
+		return nil, err
+	}
+
+	permissions := resourcePermissionsResp.GetData()
+	sensitivePermissionsMap := map[string][]string{}
+	for _, permission := range permissions {
+		for i := 0; i < len(permission.Info.OperationGroups); i++ {
+			resourceType := string(permission.Info.GetResourceType())
+			operationGroup := string(permission.Info.OperationGroups[i].GetOperationGroup())
+			tags := permission.Info.OperationGroups[i].GetTags()
+			if len(tags) == 0 || tags[0] != "SENSITIVE" {
+				continue
+			}
+			if ops, ok := sensitivePermissionsMap[resourceType]; ok {
+				ops = append(ops, operationGroup)
```

Comment:
Why do you need intermediate value ops here?

Reply:
We are iterating over all the available permissions to get the ones that are tagged as sensitive
Something like:
map[ACCOUNT_LOGIN_TYPE:[UPDATE] ACCOUNT_USER:[CREATE UPDATE DELETE] API_KEY:[REVOKE CREATE] ROLE:[CREATE UPDATE DELETE]]

I am using this map to compare the permissions that are given as input for a role or to compare the existing permissions of a role to determine whether we need the sensitive permissions confirmation

Reply:
Please lmk if I've misunderstood you

Reply:
This is not working ?
```
sensitivePermissionsMap[resourceType] =  append(sensitivePermissionsMap[resourceType],operationGroup)
```

Reply:
@harshitha-yb I agree with Gwenn. Also, the else block is not needed, if the key doesn't exist it will be an empty list anyway. So, I believe the `if else` block is not needed at all.

Reply:
Done, thanks a lot